src/Arcus.WebApi.Security/Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -63,29 +80,52 @@ public JwtTokenReader(TokenValidationParameters tokenValidationParameters, strin
         /// <param name="token">The JWT token.</param>
         public async Task<bool> IsValidTokenAsync(string token)
         {
+            bool result;
+
             try
             {
-                TokenValidationParameters validationParameters = await DetermineTokenValidationParametersAsync();
-                _handler.ValidateToken(token, validationParameters, out SecurityToken jwtToken);
+                var validationParameters = await DetermineTokenValidationParametersAsync();
+                this._handler.ValidateToken(token, validationParameters, out var validatedToken);
+                result = ValidateClaimCheck(validatedToken);
             }
-            catch (Exception exception)
+            catch
             {
                 return false;
             }
 
-            return true;
+            return result;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="validatedToken"></param>
+        /// <returns></returns>
+        private bool ValidateClaimCheck(SecurityToken validatedToken)
+        {
+            if (!this._claimCheck.Any()) 
+                return true;
+
+            var jwtToken = (JwtSecurityToken) validatedToken;
+
+            var validClaims = (from c in jwtToken.Claims
```

Comment:
What's the goal here? I'm not sure I get the logic 🤔

Reply:
He's linking the claimchecks with the claims that exist in the token.   If the number of claims found that have the key and value of what is found in the claimchecks corresponds with the number of claimchecks, all checks have passed.

Reply:
You are right @fgheysels. So basically I'm comparing the claims coming in ClaimCheck object vs the claims in the jwt object based on the claim name and value. If the result is the same number of claims the original ClaimCheck object, then all the claims are valid.